cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorRbacScopeTest.java

```diff
@@ -0,0 +1,509 @@
+        when(mockKafkaOps.get(eq(namespace), eq(clusterName))).thenReturn(kafka);
+        when(mockKafkaOps.updateStatusAsync(any(Kafka.class))).thenReturn(Future.succeededFuture());
+
+        // Mock the operations for RoleBindings
+        RoleBindingOperator mockRoleBindingOps = supplier.roleBindingOperations;
+        // Capture the names of reconciled rolebindings and their patched state
+        ArgumentCaptor<String> roleBindingNameCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<RoleBinding> roleBindingCaptor = ArgumentCaptor.forClass(RoleBinding.class);
+        when(mockRoleBindingOps.reconcile(eq(namespace), roleBindingNameCaptor.capture(), roleBindingCaptor.capture()))
+                .thenReturn(Future.succeededFuture());
+
+        KafkaAssemblyOperatorRolesSubset kao = new KafkaAssemblyOperatorRolesSubset(
+                vertx,
+                new PlatformFeaturesAvailability(false, kubernetesVersion),
+                certManager,
+                passwordGenerator,
+                supplier,
+                configNamespaceRbacScope);
+
+        Checkpoint async = context.checkpoint();
+        kao.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, namespace, clusterName))
+                .onComplete(context.succeeding(v -> context.verify(() -> {
+                    List<String> roleBindingNames = roleBindingNameCaptor.getAllValues();
+                    List<RoleBinding> roleBindings = roleBindingCaptor.getAllValues();
+
+                    assertThat(roleBindingNames, hasSize(4));
+                    assertThat(roleBindings, hasSize(4));
+
+                    // Check all RoleBindings, easier to index by order applied
+                    assertThat(roleBindingNames.get(0), is("test-instance-entity-topic-operator-role"));
+                    assertThat(roleBindings.get(0), hasRoleRef(new RoleRefBuilder()
+                            .withApiGroup("rbac.authorization.k8s.io")
+                            .withKind("Role")
+                            .withName("test-instance-entity-operator")
+                            .build()));
+
+                    assertThat(roleBindingNames.get(1), is("strimzi-test-instance-entity-topic-operator"));
+                    assertThat(roleBindings.get(1), is(nullValue()));
+
+                    assertThat(roleBindingNames.get(2), is("test-instance-entity-user-operator-role"));
+                    assertThat(roleBindings.get(2), hasRoleRef(new RoleRefBuilder()
+                            .withApiGroup("rbac.authorization.k8s.io")
+                            .withKind("Role")
+                            .withName("test-instance-entity-operator")
+                            .build()));
+
+                    assertThat(roleBindingNames.get(3), is("strimzi-test-instance-entity-user-operator"));
+                    assertThat(roleBindings.get(3), is(nullValue()));
+
+                    async.flag();
+                })));
```

Comment:
Can we also add checks here that the ClusterRoleBinding etc. APIs are never called? That would help to test not just that it does the right steps but that it skips the wrong steps.